src/dbos-executor.ts

```diff
@@ -1898,12 +1899,44 @@ export class DBOSExecutor implements DBOSExecutorContext {
     }
   }
 
-  getWorkflowStatus(workflowID: string, callerID?: string, callerFN?: number): Promise<WorkflowStatus | null> {
-    return this.systemDatabase.getWorkflowStatus(workflowID, callerID, callerFN);
+  async getWorkflowStatus(workflowID: string, callerID?: string, callerFN?: number): Promise<WorkflowStatus | null> {
+    const status = await this.systemDatabase.getWorkflowStatus(workflowID, callerID, callerFN);
+    const inputs = await this.systemDatabase.getWorkflowInputs(workflowID);
+    if (status) {
+      return {
+        workflowID: status.workflowUUID,
+        status: status.status,
+        workflowName: status.workflowName,
+        workflowClassName: status.workflowClassName,
+        workflowConfigName: status.workflowConfigName,
+        queueName: status.queueName,
+        authenticatedUser: status.authenticatedUser,
+        authenticatedRoles: status.authenticatedRoles,
+        assumedRole: status.assumedRole,
+        output: status.output ?? undefined,
+        error: status.error ? deserializeError(status.error) : undefined,
+        input: inputs ?? undefined,
+        createdAt: status.createdAt,
+        updatedAt: status.updatedAt,
+        recoveryAttempts: status.recoveryAttempts,
+        applicationID: status.applicationID,
+        applicationVersion: status.applicationVersion,
+        executorId: status.executorId,
+        request: status.request,
+      };
```

Comment:
Should we use `DBOSExecutor.toWorkflowStatus(status, true)` here?

Reply:
Resolving this convo in favor of earlier thread in the same function